common/common_functions.sh

```diff
@@ -817,6 +817,120 @@ function deploy_k8s_bare {
     fi
 }
 
+function deploy_kind_cluster_with_multus {
+    local cluster_name="$1"
+    local workers_number="$2"
+
+    deploy_kind_cluster "$cluster_name" "$workers_number"
+    let status=status+$?
+    if [ "$status" != 0 ]; then
+        echo "Failed to deploy kind!!"
+        popd
+        return $status
+    fi
+
+    deploy_multus
+    let status=status+$?
+    if [ "$status" != 0 ]; then
+        echo "Failed to deploy multus!!"
+        popd
+        return $status
+    fi
+}
+
+function deploy_kind_cluster {
+    local cluster_name="$1"
+    local workers_number="$2"
+    local kind_conf_file=${3:-"$ARTIFACTS/kind_confs.yaml"}
+
+    local status=0
+
+    prepare_kind_confs "$kind_conf_file" "$cluster_name" "$workers_number"
+    let status=$status+$?
+    if [[ $status != "0" ]];then
+        echo "Error: failed to prepare kind confs!"
+        return 1
+    fi
+
+    sudo kind create cluster --config "$kind_conf_file"
+    let status=$status+$?
+    if [[ $status != "0" ]];then
+        echo "Error: failed to create kind cluster $cluster_name!"
+        return $status
+    fi
+
+    if [[ "$workers_number" -ge '1' ]];then
+        remount_workers_sys_fs
+        let status=$status+$?
+        if [[ $status != "0" ]];then
+            echo "Error: failed to create kind cluster $cluster_name!"
+            return $status
+        fi
+    fi
+}
+
+function remount_workers_sys_fs {
+    local status=0
+
+    local worker_containers=$(sudo docker ps -f name=kind-worker -q)
+
+    if [[ -z "$worker_containers" ]];then
+        echo "Error: No worker containers found!"
+        return 1
+    fi
+
+    for worker_container in $worker_containers; do
+        sudo docker exec -it "$worker_container" mount -o remount,rw /sys
```

Comment:
Why is this needed?

Reply:
This is need to be able to create VFs, since to create them, write access to the sriov_numvfs is needed. and kind by default remount the sys fs to be read only. i think they do so to not have many container (nodes) writing to the same file.

Reply:
Is it configurable in kind to mount it with write permission?

Reply:
no, it is hard coded, see https://github.com/kubernetes-sigs/kind/blob/b6bc112522651d98c81823df56b7afa511459a3b/images/base/files/usr/local/bin/entrypoint#L147

Reply:
@ykulazhenkov  unfortunately this will not work also, since this basically mount the directory inside the container, then the entrypoint will do the remount to read only after the container is started.

P.S. I did try this the first time i faced the issue, but that did not work.